### PR TITLE
feat: Add embassy-based sleeper for no_std

### PR DESCRIPTION
**Pull request by wackazong**

### CODE REVIEW

File: backon/src/blocking_sleep.rs [hunk 1]
```diff
@@ -24,13 +24,18 @@ impl<F: Fn(Duration) + 'static> BlockingSleeper for F {
...
 /// The default implementation of `Sleeper` while feature `std-blocking-sleep` enabled.
 ///
 /// it uses [`std::thread::sleep`].
 #[cfg(feature = "std-blocking-sleep")]
 pub type DefaultBlockingSleeper = StdSleeper;
+/// The default implementation of `Sleeper` while feature `embassy-sleep` enabled.
+///
+/// it uses [`embassy_time::block_for`].
+#[cfg(all(not(feature = "std-blocking-sleep"), feature = "embassy-sleep"))]
+pub type DefaultBlockingSleeper = EmbassySleeper;
```
Comment (Xuanwo) [inline backon/src/blocking_sleep.rs:38]: Plesse don't add `EmbassySleeper` as default sleeper. Users will need to specify `EmbassySleeper` instead.

File: backon/src/sleep.rs [hunk 2]
```diff
@@ -91,3 +100,17 @@ impl Sleeper for GlooTimersSleep {
         gloo_timers::future::sleep(dur)
     }
 }
+
+/// The default implementation of `Sleeper` utilizes `embassy-time::Timer`.
+#[cfg(all(not(feature = "std"), feature = "embassy-sleep"))]
+#[derive(Clone, Copy, Debug, Default)]
+pub struct EmbassySleep;
```
Comment (Xuanwo) [inline backon/src/sleep.rs:107]: Please call it `EmbassySleeper` to align the name.

File: backon/src/sleep.rs [hunk 3]
```diff
@@ -91,3 +100,17 @@ impl Sleeper for GlooTimersSleep {
         gloo_timers::future::sleep(dur)
     }
 }
+
+/// The default implementation of `Sleeper` utilizes `embassy-time::Timer`.
+#[cfg(all(not(feature = "std"), feature = "embassy-sleep"))]
```
Comment (Xuanwo) [inline backon/src/sleep.rs:105]: embassy can't be used along with `std`?

File: backon/src/blocking_sleep.rs (hunk 1 above)
Comment (wackazong) [reply backon/src/blocking_sleep.rs]: Ok

File: backon/src/sleep.rs (hunk 2 above)
Comment (wackazong) [reply backon/src/sleep.rs]: Ok

File: backon/src/blocking_sleep.rs [hunk 4]
```diff
@@ -24,7 +24,7 @@ impl<F: Fn(Duration) + 'static> BlockingSleeper for F {
 /// The default implementation of `Sleeper` when no features are enabled.
 ///
 /// It will fail to compile if a containing [`Retry`][crate::Retry] is `.await`ed without calling [`Retry::sleep`][crate::Retry::sleep] to provide a valid sleeper.
-#[cfg(not(feature = "std-blocking-sleep"))]
+#[cfg(all(not(feature = "std-blocking-sleep"), not(feature = "embassy-sleep")))]
```
Comment (Xuanwo) [inline backon/src/blocking_sleep.rs:27]: Hi, we don't need to change this.

File: backon/src/blocking_sleep.rs [hunk 5]
```diff
@@ -53,3 +53,15 @@ impl BlockingSleeper for StdSleeper {
         std::thread::sleep(dur)
     }
 }
+
+/// A no_std blocking sleeper
+#[cfg(feature = "embassy-sleep")]
+#[derive(Clone, Copy, Debug, Default)]
+pub struct EmbassySleeper;
```
Comment (Xuanwo) [inline backon/src/blocking_sleep.rs:60]: If `embassy` can be used in both blocking and async contexts, I propose creating a new file named `embassy_timer_sleep.rs`. In this file, we would implement both `BlockingSleeper` and `Sleeper` for `EmbassyTimerSleeper`.

**Comment by wackazong** [comment]
How do you propose to use the EmbassySleeper? I thought the usage of 

`backon = { version = "x.z", default-features = false, features = ["embassy-sleep"] }` 

would automatically activate embassy-sleep for the async and the blocking sleeper. That means if the feature `embassy-sleep` is active, the DefaultSleeper and the DefaultBlockingSleeper should be the embassy sleepers as far as I can see. 

Let me know what you think.

**Comment by wackazong** [comment]
Maybe this is a possibility?

File: backon/src/embassy_timer_sleep.rs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    type Sleep = embassy_time::Timer;
+
+    fn sleep(&self, dur: Duration) -> Self::Sleep {
+        embassy_time::Timer::after_millis(dur.as_millis() as u64)
+    }
+}
+
+/// A no_std blocking sleeper based on the embassy framework (https://embassy.dev)
+#[derive(Clone, Copy, Debug, Default)]
+pub struct EmbassyBlockingSleeper;
```
Comment (Xuanwo) [inline backon/src/embassy_timer_sleep.rs:18]: We don't need `EmbassyBlockingSleeper` anymore.

File: backon/src/embassy_timer_sleep.rs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+use crate::{BlockingSleeper, Sleeper};
+use core::time::Duration;
+
+/// A no_std async sleeper based on the embassy framework (https://embassy.dev)
+#[derive(Clone, Copy, Debug, Default)]
+pub struct EmbassySleeper;
```
Comment (Xuanwo) [inline backon/src/embassy_timer_sleep.rs:6]: EmbassySleeper can implement both async and blocking sleep.

File: backon/src/lib.rs [hunk 8]
```diff
@@ -184,5 +185,12 @@ pub use blocking_sleep::DefaultBlockingSleeper;
 #[cfg(feature = "std-blocking-sleep")]
 pub use blocking_sleep::StdSleeper;
 
+#[cfg(feature = "embassy-sleep")]
+mod embassy_timer_sleep;
+#[cfg(feature = "embassy-sleep")]
+pub use embassy_timer_sleep::EmbassyBlockingSleeper;
```
Comment (Xuanwo) [inline backon/src/lib.rs:191]: We don't need this anymore.

**Review by Xuanwo — APPROVED** [review]
Thank you @wackazong for building this, really appricated!

**Pull request merged**